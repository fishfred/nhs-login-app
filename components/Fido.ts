import { NhsLogin } from "./NhsLogin";

import ReactNativeBiometrics from "react-native-biometrics";

import ReactNative, { Alert } from 'react-native';

const { FidoUAF } = ReactNative.NativeModules;
const FacetID = "android:apk-key-hash-sha256:47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU";

export class Fido {
    async register(accessToken: string, onError: (err: string) => void) {
        const result: string = await FidoUAF.FidoUafRegister({ accessToken: accessToken, facetId: FacetID, url: "https://uaf.sandpit.signin.nhs.uk/regRequest" });
        const resultJson = JSON.parse(result);
        if (resultJson.error) {
            Alert.alert("Fingerprint Error", resultJson.error,
                [
                    {
                        text: "Cancel",
                        onPress: () => undefined,
                        style: "cancel"
                    },
                    {
                        text: "Continue",
                        onPress: () => { }
                    }
                ]);
            return;
        }
        console.log(resultJson);
        const res = await fetch("https://uaf.sandpit.signin.nhs.uk/regResponse", {
            method: "POST",
            body: resultJson
        }).then((res) => res.json()).catch((err) => {
            console.log(err);
        });
        console.log(res);
    }

    async auth(): Promise<string> {
        const result: string = await FidoUAF.FidoUafAuthenticate({
            facetId: FacetID,
            url: "https://uaf.sandpit.signin.nhs.uk/authRequest"
        });

        return result;
    }

    async ToBase64Url(s: string): Promise<string> {
        return await FidoUAF.ToBase64Url({
            input: s
        });
    }
}