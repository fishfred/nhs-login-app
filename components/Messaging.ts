import {connect, Socket} from "socket.io-client";

export class Messaging {
    sio: typeof Socket
    app_token: string
    static instance: Messaging
    constructor(server_url: string, app_token: string){
        if (Messaging.instance){
            console.warn("Messaging instance already exists");
        }
        Messaging.instance = this;
        this.app_token = app_token;
        this.sio = connect(server_url, {
            transports: ["websocket"],
            // extraHeaders: {
            //     "Authorization": "Bearer " + app_token
            // }
        });
        this.sio.on("connection", () => {
            this.sio.send("hello");
        });
    }
}