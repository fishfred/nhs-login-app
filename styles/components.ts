import {StyleSheet} from 'react-native';
import * as Colors from './colors';

export const components = StyleSheet.create({
  textField: {
    borderColor: Colors.FormBorderColor,
    borderWidth: 3,
    backgroundColor: Colors.White,
    fontSize: 21,
    paddingVertical: 3,
    paddingHorizontal: 5,
  },
  picker: {
    borderColor: Colors.FormBorderColor,
    borderWidth: 3,
    borderStyle: 'solid',
    backgroundColor: Colors.White,
    fontSize: 21,
    paddingVertical: 5,
    paddingHorizontal: 5,
    marginBottom: 3,
    flexDirection: 'row',
    alignItems: 'center',
  },
  pickerText: {
    fontSize: 21,
    width: '100%',
    flexShrink: 1,
  },
  label: {
    fontSize: 20,
    fontWeight: 'bold',
    marginTop: 5,
  },
});
