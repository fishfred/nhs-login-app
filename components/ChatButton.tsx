import { Image, View, Text, StyleSheet, TouchableOpacity } from "react-native";
import React from "react";
import { ChatInfo } from "./Messaging";

import * as Colors from "../styles/colors";

export interface ChatButtonProps {
    name: string,
    location: string,
    imageSrc: string,
    id: string,
    light?: boolean
    onPress: (chat: ChatInfo) => void;
}
export interface ChatButtonState {}

export class ChatButton extends React.Component<ChatButtonProps, ChatButtonState>{
    render(){
        return (
            <TouchableOpacity style={styles.root} onPress={() => {
                this.props.onPress({
                    gp: {
                        name: this.props.name,
                        id: this.props.id,
                        image: this.props.imageSrc,
                        location: this.props.location
                    }
                })
            }}>
                <Image style={styles.roundImage} source={{uri: this.props.imageSrc}}></Image>
                    <View style={styles.details}>
                    <Text style={{...styles.title, color: this.props.light ? Colors.White : Colors.TextColor}}>{this.props.name}</Text>
                    <Text style={{...styles.subtitle, color: this.props.light ? Colors.White : Colors.TextColor}}>{this.props.location}</Text>
                </View>
            </TouchableOpacity>
        )
    }
}

const styles = StyleSheet.create({
    root: {
        flexDirection: "row",
        alignItems: "center"
    },
    light: {
        color: Colors.White
    },
    title: {
        fontSize: 19,
        fontWeight: "bold"
    },
    subtitle: {
        fontSize: 19
    },
    roundImage: {
        width: 46,
        height: 46,
        borderRadius: 23
    },
    details: {
        marginLeft: 5
    }
})