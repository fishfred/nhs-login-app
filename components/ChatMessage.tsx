import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {Message, DisplayMessage} from './Messaging';
import * as Colors from '../styles/colors';

export interface ChatMessageProps {
  message: DisplayMessage;
}

export class ChatMessage extends React.Component<ChatMessageProps, {}> {
  render() {
    return (
      <View
        style={{
          ...styles.chatMessageContainer,
          flexDirection: this.props.message.gp ? 'row-reverse' : 'row',
        }}>
        <View style={styles.padding} />
        <View
          style={
            this.props.message.gp
              ? {...styles.chatMessageBox, ...styles.chatMessageBoxGp}
              : {...styles.chatMessageBox, ...styles.chatMessageBoxPatient}
          }>
          <Text
            style={
              this.props.message.gp
                ? styles.chatMessageTextGp
                : styles.chatMessageTextPatient
            }>
            {this.props.message.data}
          </Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  chatMessageContainer: {
    display: 'flex',
  },
  padding: {
    width: '100%',
    flexShrink: 1,
  },
  chatMessageBox: {
    paddingVertical: 7,
    marginVertical: 3,
    flexShrink: 0,
    paddingHorizontal: 30,
    borderRadius: 300,
  },
  chatMessageBoxPatient: {
    backgroundColor: Colors.Green,
    borderBottomRightRadius: 3,
  },
  chatMessageBoxGp: {
    borderWidth: 1,
    borderColor: Colors.BorderColor,
    borderBottomLeftRadius: 3,
  },
  chatMessageTextPatient: {
    fontSize: 20,
    color: Colors.White,
  },
  chatMessageTextGp: {
    fontSize: 20,
  },
});
