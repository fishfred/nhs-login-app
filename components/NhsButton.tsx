import React from 'react';
import { Button, StyleSheet, View, Text, TouchableWithoutFeedback, ViewStyle, GestureResponderEvent } from 'react-native';
import * as Colors from '../styles/colors';

import {SetScopes} from './NhsLogin';

type NhsButtonStyle = 'primary' | 'secondary';

export interface NhsButtonProps {
    style: NhsButtonStyle,
    text: string,
    onPress: (event: GestureResponderEvent) => void;
}

interface NhsButtonState {
    currentlyPressed: boolean;
}

export class NhsButton extends React.Component<NhsButtonProps, NhsButtonState> {
    constructor(props: NhsButtonProps){
        super(props);
        this.state = {
            currentlyPressed: false
        }
    }

    onPressIn(){
        this.setState({
            currentlyPressed: true
        });
    }

    onPressOut(){
        setTimeout(() => {
            this.setState({
                currentlyPressed: false
            });
        }, 200);
    }

    render() {
        return (
            <TouchableWithoutFeedback accessibilityLabel={this.props.text} accessibilityRole="button" onPress={this.props.onPress} onPressIn={() => this.onPressIn()} onPressOut={() => this.onPressOut()}>
                <View style={styles(this.props.style, this.state.currentlyPressed).container}>
                    <View style={[styles(this.props.style, this.state.currentlyPressed).front]}>
                        <View style={styles(this.props.style, this.state.currentlyPressed).behind}></View>
                    </View>
                    <View style={styles(this.props.style, this.state.currentlyPressed).button}>
                        <Text style={styles(this.props.style, this.state.currentlyPressed).buttonText}>{this.props.text}</Text>
                    </View>
                </View>
            </TouchableWithoutFeedback>
        );
    }
}

const SHADOW_HEIGHT = 4;
const HEIGHT = 53;
const BORDER_RADIUS = 4;

function styles(buttonStyle: NhsButtonStyle, currentlyPressed: boolean): any {
    return StyleSheet.create({
        container: {
            height: HEIGHT + SHADOW_HEIGHT,
            marginTop: 5,
            // flex: 1,
            alignItems: 'center'
        },
        button: {
            width: "100%",
            height: HEIGHT,
            textTransform: 'none',
            backgroundColor: buttonStyle == 'primary' ? Colors.ButtonColor : Colors.SecondaryButtonColor,
            padding: 14,
            borderRadius: BORDER_RADIUS,
            marginTop: currentlyPressed ? SHADOW_HEIGHT : 0
        },
        front: {
            width: "100%",
            height: HEIGHT,
            alignItems: 'center',
            justifyContent: 'center',
            zIndex: -1,
            position: 'absolute',
            left: 0,
            top: 0
        },
        behind: {
            alignItems: 'center',
            justifyContent: 'center',
            position: 'absolute',
            left: 0,
            top: 0,
            width: "100%",
            marginTop: SHADOW_HEIGHT,
            height: HEIGHT,
            backgroundColor: "#414141",
            zIndex: 3,
            borderRadius: BORDER_RADIUS
        },
        buttonText: {
            textAlign: 'center',
            color: Colors.ButtonTextColor,
            fontSize: 19,
            fontWeight: 'bold'
        }
    })
}
