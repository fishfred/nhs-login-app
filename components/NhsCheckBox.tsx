import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
//@ts-ignore
import CheckMark from '../assets/icons/checkbox.svg';
import * as Colors from '../styles/colors';

export interface NhsCheckBoxState {
    disabled: boolean,
    value: boolean
}

export class NhsCheckBox extends React.Component<NhsCheckBoxState> {
    render() {
        return (
            <View style={styles.box}>
                {this.props.value ? <CheckMark></CheckMark> : undefined}
            </View>
        )
    }
}

const styles = StyleSheet.create({
    box: {
        borderColor: Colors.FormBorderColor,
        borderWidth: 2,
        height: 40,
        width: 40,
        marginRight: 10
    }
});