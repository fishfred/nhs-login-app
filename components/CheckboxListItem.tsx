import CheckBox from '@react-native-community/checkbox';
import React from 'react';
import { Pressable, StyleSheet, Text, View } from 'react-native';
import { TouchableHighlight, TouchableOpacity, TouchableWithoutFeedback } from 'react-native-gesture-handler';
import { onChange } from 'react-native-reanimated';
import { NhsCheckBox } from './NhsCheckBox';

export interface CheckBoxListItemProps {
    title: string;
    checked: boolean;
    disabled: boolean;
    onChange: (checked: boolean) => void
}


export class CheckBoxListItem extends React.Component<CheckBoxListItemProps> {
    render() {
        return (
            <TouchableWithoutFeedback accessible={true}
                accessibilityRole="checkbox"
                accessibilityLabel={"scope " + this.props.title}
                // accessibilityHint={`Toggles the scope ${this.props.title}`}
                accessibilityState={{ checked: this.props.checked, disabled: this.props.disabled }}
                disabled={this.props.disabled}
                style={[style.view, { opacity: this.props.disabled ? 0.4 : 1 }]}
                onPress={() => {
                    this.props.onChange(!this.props.checked);
                }}>
                <NhsCheckBox value={this.props.checked} disabled={this.props.disabled}></NhsCheckBox><Text style={style.text}>{this.props.title}</Text>
            </TouchableWithoutFeedback>
        )
    }
}

const style = StyleSheet.create({
    view: {
        display: 'flex',
        flexDirection: 'row',
        height: 50,
        alignItems: 'center',
        paddingLeft: 30
    },
    checkbox: {
        marginRight: 10,
        width: 25,
        height: 25
    },
    text: {
        fontSize: 20
    }
})