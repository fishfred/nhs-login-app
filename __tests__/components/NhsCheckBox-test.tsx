import React from 'react';
import { NhsCheckBox } from "../../components/NhsCheckBox";
import renderer from 'react-test-renderer';

describe("<NhsCheckBox>", () => {
it('displays no check when value=false', () => {
    const tree = renderer.create(<NhsCheckBox value={false} disabled={false} />);
    tree.root.findByType(NhsCheckBox).instance.setState();
    expect(tree).toMatchSnapshot();
});
it('displays a check when value=true', () => {
    const tree = renderer.create(<NhsCheckBox value={true} disabled={false} />);
    tree.root.findByType(NhsCheckBox).instance.setState();
    expect(tree).toMatchSnapshot();
});
});