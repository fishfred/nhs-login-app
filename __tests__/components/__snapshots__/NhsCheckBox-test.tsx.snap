// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NhsCheckBox> displays a check when value=true 1`] = `
<View
  style={
    Object {
      "borderColor": "#4c6272",
      "borderWidth": 2,
      "height": 40,
      "marginRight": 10,
      "width": 40,
    }
  }
>
  <SvgMock />
</View>
`;

exports[`<NhsCheckBox> displays no check when value=false 1`] = `
<View
  style={
    Object {
      "borderColor": "#4c6272",
      "borderWidth": 2,
      "height": 40,
      "marginRight": 10,
      "width": 40,
    }
  }
/>
`;
